src/IonTimestamp.ts

```diff
@@ -310,154 +281,173 @@ export class Timestamp {
         */
     }
 
-  private checkValid() : void {
-    if (this.precision === Precision.NULL) {
-      return;
-    }
-
-    if (this.offset < MIN_OFFSET || this.offset > MAX_OFFSET) {
-      throw new Error(`Offset ${this.offset} must be between ${MIN_OFFSET} and ${MAX_OFFSET} inclusive`);
-    }
-
-    switch (this.precision) {
-      default:
-        throw new Error(`Unknown precision ${this.precision}`);
-      case Precision.SECONDS:
-        let seconds: number = this.seconds.numberValue();
-        if (seconds < MIN_SECONDS || seconds >= MAX_SECONDS) {
-          throw new Error(`Seconds ${seconds} must be between ${MIN_SECONDS} inclusive and ${MAX_SECONDS} exclusive`);
-        }
-      case Precision.HOUR_AND_MINUTE:
-        if (this.minute < MIN_MINUTE || this.minute > MAX_MINUTE) {
-          throw new Error(`Minute ${this.minute} must be between ${MIN_MINUTE} and ${MAX_MINUTE} inclusive`);
+    private checkValid() : void {
+        if (this.precision === Precision.NULL) {
+            return;
         }
-        if (this.hour < MIN_HOUR || this.hour > MAX_HOUR) {
-          throw new Error(`Hour ${this.hour} must be between ${MIN_HOUR} and ${MAX_HOUR} inclusive`);
-        }
-      case Precision.DAY:
-        if (this.day < MIN_DAY || this.day > MAX_DAY) {
-          throw new Error(`Day ${this.day} must be between ${MIN_DAY} and ${MAX_DAY} inclusive`);
+
+        if (this.offset < MIN_OFFSET || this.offset > MAX_OFFSET) {
+            throw new Error(`Offset ${this.offset} must be between ${MIN_OFFSET} and ${MAX_OFFSET} inclusive`);
         }
-      case Precision.MONTH:
-        if (this.month < MIN_MONTH || this.month > MAX_MONTH) {
-          throw new Error(`Month ${this.month} must be between ${MIN_MONTH} and ${MAX_MONTH} inclusive`);
+
+        switch (this.precision) {
+            default:
```

Comment:
`default` should be the last switch case.

Another thought:  is having an offset valid for values that have a timestamp precision that is less than `HOUR_AND_MINUTE`?  Perhaps not.